### PR TITLE
Add normalize_whitepace filter

### PR DESCRIPTION
**Pull request by pathawks**
See #4877

### CODE REVIEW

**Comment by envygeeks** [comment]
Maybe we should also add `strip_whitespace` that does `\s+|\n+` to " "?

File: site/_docs/templates.md [hunk 1]
```diff
@@ -258,6 +258,17 @@ common tasks easier.
     </tr>
     <tr>
       <td>
+        <p class="name"><strong>Normalize Whitespace</strong></p>
+        <p>Replace any occurance of whitespace with a single space.</p>
+      </td>
+      <td class="align-center">
+        <p>
+         <code class="filter">{% raw %}{{ "a \n b" | normalize_whitepace }}{% endraw %}</code>
```
Comment (parkr) [inline site/_docs/templates.md]: Is it possible to add the resulting output?

**Comment by parkr** [comment]
This looks more like a `strip_newlines` kind of change.

**Comment by pathawks** [comment]
The problem is that `strip_newlines` will improperly concatenate words.

```text
cat
dog
```

becomes

```text
catdog
```

I would not mind just changing `strip_newlines` to behave this way, but others prefer the current behavior of `strip_newlines` (it does just what it says).

**Comment by parkr** [comment]
@pathawks Ok! It's a new filter so I'm fine with it.

@jekyllbot: merge +minor

**Pull request merged**